scripts/TestGen.hx

```diff
@@ -0,0 +1,380 @@
+/*
+
+This test generator was initially written by David Bruce: https://github.com/davidbruce
+
+His initial PR was stale and his commits were accidentally nuked, hence this attribution.
+
+*/
+
+import haxe.Template;
+import haxe.Json;
+import sys.io.File;
+import sys.FileSystem;
+import sys.Http;
+
+using StringTools;
+
+class TestGen
+{
+	static var exercise:String;
+	static var exerciseStubTmpl = new Template("package;
+
+class ::exercise:: {
+    ::methods:: 
+}");
+
+	static var methodsTmpl = new Template("public static function ::method::(::args::): ::returnType:: {
+        throw \"Not Implemented\"; // Delete this statement and write your own implementation.
+    }");
+
+	static var testTmpl = new Template("package;
+
+using buddy.Should;
+
+//Created by testgen.hx
+class Test extends buddy.SingleSuite {
+    public function new() {
+        ::stories::
+    }
+}");
+
+	static var storyTmpl = new Template("describe(\"::story::\", {
+        ::testCases::});");
+
+	static var testCaseTmpl = new Template("    it(\"::testCase::\", {
+                ::testCriteria::
+            });
+        ");
+
+	public static function main()
+	{
+		// TODO: Configlet is able to create the directories/files we need:
+		//   ./bin/configlet sync --docs --filepaths --metadata --tests include -uye <slug>
+		// 
+		// We should try to change the code below to use configlet's
+		// cached copy of the problem-specifications repo instead
+		var dataUrl = new Template("https://raw.githubusercontent.com/exercism/problem-specifications/master/exercises/::exercise::/canonical-data.json");
+		var descriptionUrl = new Template("https://raw.githubusercontent.com/exercism/problem-specifications/master/exercises/::exercise::/description.md");
+		var metaUrl = new Template("https://raw.githubusercontent.com/exercism/problem-specifications/master/exercises/::exercise::/metadata.toml");
+		// check if no args are passed
+		if (Sys.args().length < 1)
+		{
+			Sys.println("An exercise is required.");
+			return;
+		}
+
+		exercise = Sys.args()[0];
+
+		// pull in files from github
+		var execute = (template:Template) -> return template.execute({exercise: exercise});
+
+		var dataJson = Http.requestUrl(execute(dataUrl));
+		var descriptionMd = Http.requestUrl(execute(descriptionUrl));
+		var metaToml = Http.requestUrl(execute(metaUrl));
+
+		var exercisesDir = "../exercises/practice/";
+		var dir = '${exercisesDir}${exercise}/';
+
+		// create exercise folder
+		if (!FileSystem.exists(dir))
+		{
+			FileSystem.createDirectory(dir);
+		}
+		// create .meta
+		if (!FileSystem.exists('${dir}.meta'))
+		{
+			FileSystem.createDirectory('${dir}.meta');
+		}
+		// create src folder
+		if (!FileSystem.exists('${dir}src'))
+		{
+			FileSystem.createDirectory('${dir}src');
+		}
+		// create test folder
+		if (!FileSystem.exists('${dir}test'))
+		{
+			FileSystem.createDirectory('${dir}test');
+		}
+
+		// create string for tests.toml; first line is always [canonical-tests]
```

Comment:
configlet sync will give us a generated test file. It'll consist of about ten one-line comments, a newline, and then alternating test GUIDs followed by a description. There might be a few additional lines of text but each test ends with a newline before the next one.

Here are some examples:

```
[8b2c43ac-7257-43f9-b552-7631a91988af]
description = "foo"

[180a8ff9-5b94-43fc-9db1-d46b4a8c93b6]
description = "bar"
include = false
comment = "optional comment"

[33eb6f87-0498-4ccf-9573-7f8c3ce92b7b]
description = "baz"
include = false

[c125dab7-2a53-492f-a99a-56ad511940d8]
description = "foobar1"
include = false

[a0c7b9b8-0e89-47f8-8b4a-c50f885e79d1]
description = "foobar2"
include = false
reimplements = "c125dab7-2a53-492f-a99a-56ad511940d8"

[d7982c4f-1602-49f6-a651-620f2614243a]
description = "foobar3"
reimplements = "a0c7b9b8-0e89-47f8-8b4a-c50f885e79d1"
```

I'd expect foo and foobar3 to be added but the others to be skipped.